lib/galaxy/model/__init__.py

```diff
@@ -5089,18 +5095,18 @@ def __init__(self, state=None, path=None, info=None, pid=None, socket=None, para
         self.params = params
 
 
-class Tag(object):
+class Tag(RepresentById):
     def __init__(self, id=None, type=None, parent_id=None, name=None):
         self.id = id
         self.type = type
         self.parent_id = parent_id
         self.name = name
 
     def __str__(self):
-        return "Tag(id=%s, type=%i, parent_id=%s, name=%s)" % (self.id, self.type, self.parent_id, self.name)
+        return "Tag(id=%s, type=%i, parent_id=%s, name=%s)" % (self.id, self.type or -1, self.parent_id, self.name)
 
 
-class ItemTagAssociation(Dictifiable):
+class ItemTagAssociation(Dictifiable, RepresentById):
```

Comment:
That's a base class as `BaseJobMetric` and `ItemRatingAssociation`, so I think this can be reverted.